lambdas/src/apis/collections/controllers/collections.ts

```diff
@@ -11,92 +17,143 @@ import {
+    const lambdasUrl = `https://${req.headers.host}/lambdas/collections/contents/${urn}`
+    const image = lambdasUrl + '/image'
+    const thumbnail = lambdasUrl + '/thumbnail'
+    const bodyShapeTraits: ERC721StandardTrait[] = getBodyShapes(wearableMetadata.data.representations).map(
+      (bodyShape) => ({
+        trait_type: 'Body Shape',
+        value: bodyShape
+      })
+    )
+
+    const tagTraits: ERC721StandardTrait[] = wearableMetadata.data.tags.map((tag) => ({
+      trait_type: 'Tag',
+      value: tag
+    }))
+
+    const standardErc721 = {
+      id: urn,
+      name,
+      description,
+      language: 'en-US',
+      image,
+      thumbnail,
+      attributes: [
+        {
+          trait_type: 'Rarity',
+          value: wearableMetadata.rarity
+        },
+        {
+          trait_type: 'Category',
+          value: wearableMetadata.data.category
         },
-        []
-      )
-
-      const tagTraits = wearableMetadata.data.tags.reduce((tags: ERC721StandardTrait[], tag) => {
-        tags.push({
-          trait_type: 'Tag',
-          value: tag
-        })
-
-        return tags
-      }, [])
-
-      const standardErc721 = {
-        id: urn,
-        name,
-        description,
-        language: 'en-US',
-        image,
-        thumbnail,
-        attributes: [
-          {
-            trait_type: 'Rarity',
-            value: wearableMetadata.rarity
-          },
-          {
-            trait_type: 'Category',
-            value: wearableMetadata.data.category
-          },
-          ...tagTraits,
-          ...bodyShapeTraits
-        ]
-      }
-      res.send(standardErc721)
-    } else {
-      res.status(404).send()
+        ...tagTraits,
+        ...bodyShapeTraits
+      ]
     }
+    res.send(standardErc721)
   } catch (e) {
-    res.status(500).send(e.message)
+    if (e instanceof ServiceError) {
+      res.status(e.statusCode).send(e.message)
+      return
+    }
+    res.status(500)
   }
 }
 
-export async function contentsImage(client: SmartContentClient, req: Request, res: Response): Promise<void> {
+export async function contentsThumbnail(
+  client: SmartContentClient,
+  req: Request,
+  res: Response,
+  storage: string
+): Promise<void> {
   // Method: GET
-  // Path: /contents/:urn/image
-  const { urn } = req.params
-  await internalContents(client, res, urn, (wearableMetadata) => wearableMetadata.image)
+  // Path: /contents/:urn/thumbnail/:size
+  await handleImageRequest(client, req, res, storage, true)
 }
 
-export async function contentsThumbnail(client: SmartContentClient, req: Request, res: Response): Promise<void> {
+export async function contentsImage(
+  client: SmartContentClient,
+  req: Request,
+  res: Response,
+  storage: string
+): Promise<void> {
   // Method: GET
-  // Path: /contents/:urn/thumbnail
+  // Path: /contents/:urn/image/:size
+  await handleImageRequest(client, req, res, storage, false)
+}
+
+async function handleImageRequest(
```

Comment:
Should we move the logic for generating and storing images to another file?

Reply:
yes, makes sense to move image cache logic to other file 👍